src/config-utils.ts

```diff
@@ -0,0 +1,157 @@
+  if (!configPath) {
+    return defaultConfig;
+  }
+
+  const configFilePath = `${configPath}/${CONFIG_FILE_NAME}`;
+
+  if (!fs.existsSync(configFilePath)) {
+    globalConfigCache = defaultConfig;
+    return defaultConfig;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const config = require(configFilePath);
+
+  if (!config.default) {
+    config.default = defaultConfig.default;
+  }
+  validateConfig(config);
+  globalConfigCache = config;
+  return config;
+}
+
+export function getModuleConfigEntry(modulePath: string): ConfigEntry {
+  const moduleName = getModuleNameForPath(modulePath);
+  if (moduleConfigCache[moduleName]) {
+    return moduleConfigCache[moduleName];
+  }
+  const packageJson = JSON.parse(
+    fs.readFileSync(`${cwd}/${modulePath}/package.json`).toString()
+  );
+
+  // a command override is found in the package
+  if (!packageJson['watch-module']) {
+    const globalConfig = getConfig();
+    const moduleConfig = globalConfig[moduleName] || globalConfig.default;
+    moduleConfigCache[moduleName] = moduleConfig;
+    return moduleConfig;
+  }
+
+  const moduleConfig = packageJson['watch-module'];
+  if (typeof moduleConfig.includes === 'undefined') {
+    moduleConfig.includes = ['src'];
+  }
+  moduleConfigCache[moduleName] = moduleConfig;
+  return moduleConfig;
+}
+
+export function getIncludesPaths(modulePaths: string[]): string[] {
+  const srcPaths: string[] = [];
+
+  for (let i = 0; i < modulePaths.length; i++) {
+    const path = modulePaths[i];
+
+    const moduleConfig = getModuleConfigEntry(path);
+    const moduleIncludes =
+      moduleConfig.includes && Array.isArray(moduleConfig.includes)
+        ? moduleConfig.includes
+        : [];
+
+    moduleIncludes.forEach(includePath => {
+      let pathToAdd = `${path}/`;
+      if (includePath && includePath !== '.') {
+        pathToAdd = `${pathToAdd}${includePath}`;
+      }
+      if (!fs.existsSync(pathToAdd)) {
+        return;
+      }
+      srcPaths.push(pathToAdd);
+    });
+  }
+  return srcPaths;
+}
+
+export function getExcludesPaths(modulePaths: string[]): string[] {
+  const srcPaths: string[] = [];
+  for (let i = 0; i < modulePaths.length; i++) {
+    const path = modulePaths[i];
+
+    const moduleConfig = getModuleConfigEntry(path);
+
+    const moduleExcludes =
+      moduleConfig.excludes && Array.isArray(moduleConfig.excludes)
+        ? moduleConfig.excludes
+        : [];
+
+    moduleExcludes.forEach(excludePath => {
+      let pathToAdd = `${path}/`;
+      if (excludePath && excludePath !== '.') {
+        pathToAdd = `${pathToAdd}${excludePath}`;
+      }
+      if (!fs.existsSync(pathToAdd)) {
+        return;
+      }
+      srcPaths.push(pathToAdd);
+    });
+  }
+  return srcPaths;
+}
+
+export function createDefaultConfig(): void {
+  const configPath = getConfigPath();
+  if (!configPath) {
+    return;
+  }
+
+  const configFilePath = `${configPath}/${CONFIG_FILE_NAME}`;
```

Comment:
Apparently, the `.config` is filled with directory, so `~/.config/watch-module/config.json` seems more relevant, no ?

Reply:
Yep